### ISSUE TITLE
rtpengine crash with libcrypto segfault

### ISSUE THREAD

**Issue by SipSeb** [sipwise/rtpengine#419]
Hi, I know this sounds a bit like #35 and #51, but since those two bug reports were from 3 years ago and the rtpengine version is a lot newer now, I thought I better open a new bug report.

We have one rtpengine handling only WebRTC and SRTP traffic, converting it to standard RTP. And it currently crashes about twice per week with a libcrypto segfault. 

This is the backtrace of the last crash (`bt full` and `thread apply all bt`):

```
GNU gdb (Debian 7.7.1+dfsg-5) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/sbin/rtpengine...Reading symbols from /usr/lib/debug//usr/sbin/rtpengine...done.
done.
[New LWP 14657]
[New LWP 14660]
[New LWP 14661]
[New LWP 14663]
[New LWP 14662]
[New LWP 14651]
[New LWP 14658]
[New LWP 14659]
[New LWP 14653]
[New LWP 14665]
[New LWP 14664]
[New LWP 14656]
[New LWP 14652]
[New LWP 14655]
[New LWP 14650]
[New LWP 14654]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `/usr/sbin/rtpengine --config-file=/etc/rtpengine/rtpengine.conf --config-sectio'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f9c5ad43aa5 in ?? () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
#0  0x00007f9c5ad43aa5 in ?? () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
No symbol table info available.
#1  0x00007f9c5adb4b25 in ?? () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
No symbol table info available.
#2  0x00007f9c5adaf775 in EVP_EncryptUpdate () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
No symbol table info available.
#3  0x0000000000423ebd in aes_ctr_128 (out=<optimized out>, in=<optimized out>, ecc=0x7f9c2401a800, iv=iv@entry=0x7f9c52925210 "\336\323\332\r=R\233\305(Q\271\256\305\311") at crypto.c:232
        ivx = "\336\323\332\r=R\233\305(Q\271\256\305\311\000\b"
        key_block = "!\304\367\316=3\204\006\027x<\200\370\332\234\262"
        p = <optimized out>
        q = <optimized out>
        left = <optimized out>
        outlen = 16
        i = <optimized out>
        pi = 0x7f9c5292548c
        qi = 0x7f9c5292548c
        ki = 0x7f9c529251c0
        __PRETTY_FUNCTION__ = "aes_ctr_128"
#4  0x0000000000424606 in aes_cm_encrypt (c=<optimized out>, ssrc=<optimized out>, s=<optimized out>, idx=<optimized out>) at crypto.c:369
        iv = "\336\323\332\r=R\233\305(Q\271\256\305\311\000"
        ivi = 0x7f9c52925210
#5  0x0000000000424e73 in crypto_encrypt_rtp (index=18232, payload=0x7f9c52925250, rtp=<optimized out>, c=0x7f9c340356f0) at crypto.h:119
No locals.
#6  rtp_avp2savp (s=0x7f9c52925320, c=0x7f9c340356f0) at rtp.c:141
        rtp = 0x7f9c52925400
        payload = {
          s = 0x7f9c5292540c "\a0\227\364*\240\314\211\275\305\264\311\302c$p\215\260\267g\253\237^s\340\327\365\324\061&\216\203gPC3\333\025\360:h\354\334\342\245\220\267\317\236\020\232\336\232\360 \214\274r\324ym/r\265\266\376\035)^\341\360\065{\303M\345s\273d\322x\v4\310d\336\230\026\r\377x", len = 160}
        to_auth = {s = 0x7f9c34035738 "\336\323\332\r\021\071\301\262(Q\271\256\202\361V\002\347AK\t/+P|\301\230+{=\330K\242\064\"", len = 14}
        index = 18232
        ret = <optimized out>
#7  0x000000000043827d in stream_packet (tv=0x7f9c52925330, fsin=0x7f9c52925340, s=0x7f9c52925320, sfd=<optimized out>) at media_socket.c:1247
        out_srtp = <optimized out>
        call = <optimized out>
        rtp_h = 0x7f9c52925400
        handler_ret = <optimized out>
        muxed_rtcp = <optimized out>
        rtcp = <optimized out>
        unk = 0
        in_srtp = 0x7f9c34035120
        i = 8
        sink = 0x7f9c34035600
        update = 0
        endpoint = {address = {family = 0x651c00 <__socket_families>, u = {ipv4 = {s_addr = 1984760537}, ipv6 = {__in6_u = {__u6_addr8 = "\331\nMv", '\000' <repeats 11 times>, __u6_addr16 = {2777, 30285, 0, 0, 0, 0, 0, 0}, __u6_addr32 = {
                    1984760537, 0, 0, 0}}}}}, port = 19648}
        rwf_in = <optimized out>
        rwf_out = 0x436810 <call_avp2savp_rtp>
        stream = 0x7f9c34035120
        ret = <optimized out>
        stun_ret = 0
        cm = <optimized out>
        media = <optimized out>
        rtp_s = <optimized out>
#8  stream_fd_readable (fd=<optimized out>, p=<optimized out>, u=<optimized out>) at media_socket.c:1461
        sfd = <optimized out>
        buf = '\000' <repeats 128 times>, "\200\210G8\000\000\000\240,kZw\a0\227\364*\240\314\211\275\305\264\311\302c$p\215\260\267g\253\237^s\340\327\365\324\061&\216\203gPC3\333\025\360:h\354\334\342\245\220\267\317\236\020\232\336\232\360 \214\274r\324y"...
        ret = <optimized out>
        iters = <optimized out>
        update = <optimized out>
        ca = <optimized out>
        s = {s = 0x7f9c52925400 "\200\210G8", len = 172}
        ep = {address = {family = 0x651c00 <__socket_families>, u = {ipv4 = {s_addr = 2001537753}, ipv6 = {__in6_u = {__u6_addr8 = "\331\nMw", '\000' <repeats 11 times>, __u6_addr16 = {2777, 30541, 0, 0, 0, 0, 0, 0}, __u6_addr32 = {
                    2001537753, 0, 0, 0}}}}}, port = 27630}
        tv = {tv_sec = 1510394195, tv_usec = 879769}
#9  0x000000000040b0ee in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:348
        ret = 1
        i = <optimized out>
        it = 0x7f9c1400d980
        evs = {{events = 1, data = {ptr = 0x2d, fd = 45, u32 = 45, u64 = 45}}, {events = 1, data = {ptr = 0x17, fd = 23, u32 = 23, u64 = 23}}, {events = 1, data = {ptr = 0x1f, fd = 31, u32 = 31, u64 = 31}}, {events = 0, data = {ptr = 0x0,
              fd = 0, u32 = 0, u64 = 0}} <repeats 95 times>, {events = 0, data = {ptr = 0x4300000000, fd = 0, u32 = 0, u64 = 287762808832}}, {events = 0, data = {ptr = 0x450, fd = 1104, u32 = 1104, u64 = 1104}}, {events = 16, data = {
              ptr = 0x4000000000, fd = 0, u32 = 0, u64 = 274877906944}}, {events = 32668, data = {ptr = 0x2, fd = 2, u32 = 2, u64 = 2}}, {events = 93, data = {ptr = 0x100000000, fd = 0, u32 = 0, u64 = 4294967296}}, {events = 0, data = {
              ptr = 0x20000, fd = 131072, u32 = 131072, u64 = 131072}}, {events = 4096, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 0, data = {ptr = 0x7c, fd = 124, u32 = 124, u64 = 124}}, {events = 131072, data = {
              ptr = 0x100000000000, fd = 0, u32 = 0, u64 = 17592186044416}}, {events = 0, data = {ptr = 0x7f9c2c001c00, fd = 738204672, u32 = 738204672, u64 = 140308729830400}}, {events = 738204656, data = {ptr = 0x20000007f9c, fd = 32668,
              u32 = 32668, u64 = 2199023288220}}, {events = 0, data = {ptr = 0x7f9c2c000020, fd = 738197536, u32 = 738197536, u64 = 140308729823264}}, {events = 51, data = {ptr = 0x2c001ba000000000, fd = 0, u32 = 0,
              u64 = 3170564511677546496}}, {events = 32668, data = {ptr = 0x7f9c58fa52af, fd = 1492800175, u32 = 1492800175, u64 = 140309484425903}}, {events = 512, data = {ptr = 0x2c00002000000000, fd = 0, u32 = 0,
              u64 = 3170534275107782656}}, {events = 32668, data = {ptr = 0x1f0, fd = 496, u32 = 496, u64 = 496}}, {events = 512, data = {ptr = 0x3300000000, fd = 0, u32 = 0, u64 = 219043332096}}, {events = 0, data = {ptr = 0x7f9c58fa64a4,
              fd = 1492804772, u32 = 1492804772, u64 = 140309484430500}}, {events = 1385331712, data = {ptr = 0x2000007f9c, fd = 32668, u32 = 32668, u64 = 137438986140}}, {events = 0, data = {ptr = 0x199ead8, fd = 26864344, u32 = 26864344,
              u64 = 26864344}}, {events = 1492811714, data = {ptr = 0x800007f9c, fd = 32668, u32 = 32668, u64 = 34359771036}}, {events = 0, data = {ptr = 0x7f9c5b799ba8, fd = 1534696360, u32 = 1534696360, u64 = 140309526322088}}, {
            events = 738204672, data = {ptr = 0x42d9410000007f9c, fd = 32668, u32 = 32668, u64 = 4816952744705294236}}, {events = 4031211391, data = {ptr = 0x7f9c2c0008c0, fd = 738199744, u32 = 738199744, u64 = 140308729825472}}, {
            events = 24, data = {ptr = 0x3300000000, fd = 0, u32 = 0, u64 = 219043332096}}, {events = 0, data = {ptr = 0x7f9c2c0008e0, fd = 738199776, u32 = 738199776, u64 = 140308729825504}}, {events = 51, data = {ptr = 0x5b79abc300000000,
              fd = 0, u32 = 0, u64 = 6591488383640403968}}, {events = 32668, data = {ptr = 0x33, fd = 51, u32 = 51, u64 = 51}}, {events = 16, data = {ptr = 0x2000000000, fd = 0, u32 = 0, u64 = 137438953472}}, {events = 0, data = {
              ptr = 0x7f9c40001ce0, fd = 1073749216, u32 = 1073749216, u64 = 140309065374944}}}
        e = {events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}
#10 0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
        p = 0x19be970
#11 0x000000000040b500 in thread_detach_func (d=0x19a1420) at aux.c:143
        dt = 0x19a1420
        t = 0x7f9c40001460
#12 0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
No symbol table info available.
#13 0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6
No symbol table info available.

Thread 16 (Thread 0x7f9c5412b700 (LWP 14654)):
#0  0x00007f9c59012c03 in epoll_wait () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x000000000040af6b in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:302
#2  0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
#3  0x000000000040b500 in thread_detach_func (d=0x19a1450) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 15 (Thread 0x7f9c5bf44980 (LWP 14650)):
#0  0x00007f9c58fe3f2d in nanosleep () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00007f9c5900bfb4 in usleep () from /lib/x86_64-linux-gnu/libc.so.6
#2  0x0000000000408412 in main (argc=1, argv=0x7ffcbcaa8908) at main.c:690

Thread 14 (Thread 0x7f9c5392a700 (LWP 14655)):
#0  0x00007f9c59012c03 in epoll_wait () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x000000000040af6b in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:302
#2  0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
#3  0x000000000040b500 in thread_detach_func (d=0x19a1440) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 13 (Thread 0x7f9c5512d700 (LWP 14652)):
#0  0x00007f9c58fe3f2d in nanosleep () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00007f9c5900bfb4 in usleep () from /lib/x86_64-linux-gnu/libc.so.6
#2  0x000000000040b45b in poller_timer_loop (d=0x19be970) at poller.c:489
#3  0x000000000040b500 in thread_detach_func (d=0x19a1470) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 12 (Thread 0x7f9c53129700 (LWP 14656)):
#0  0x00007f9c59012c03 in epoll_wait () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x000000000040af6b in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:302
#2  0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
#3  0x000000000040b500 in thread_detach_func (d=0x19a1430) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 11 (Thread 0x7f9c2affd700 (LWP 14664)):
#0  0x00007f9c59012c03 in epoll_wait () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x000000000040af6b in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:302
#2  0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
#3  0x000000000040b500 in thread_detach_func (d=0x19a1380) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 10 (Thread 0x7f9c2a7fc700 (LWP 14665)):
#0  0x00007f9c59012c03 in epoll_wait () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x000000000040af6b in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:302
#2  0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
#3  0x000000000040b500 in thread_detach_func (d=0x19a1370) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 9 (Thread 0x7f9c5492c700 (LWP 14653)):
#0  0x00007f9c592e13f8 in pthread_cond_timedwait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x0000000000435078 in __cond_timedwait_tv (tv=<synthetic pointer>, m=0x652840 <ice_agents_timers_lock>, c=0x652800 <ice_agents_timers_cond>) at aux.h:283
#2  ice_thread_run (p=<optimized out>) at ice.c:1363
#3  0x000000000040b500 in thread_detach_func (d=0x19a1460) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 8 (Thread 0x7f9c51926700 (LWP 14659)):
#0  0x00007f9c59012c03 in epoll_wait () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x000000000040af6b in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:302
#2  0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
#3  0x000000000040b500 in thread_detach_func (d=0x19a1400) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 7 (Thread 0x7f9c52127700 (LWP 14658)):
#0  0x00007f9c59012c03 in epoll_wait () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x000000000040af6b in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:302
#2  0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
#3  0x000000000040b500 in thread_detach_func (d=0x19a1410) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 6 (Thread 0x7f9c5592e700 (LWP 14651)):
#0  0x00007f9c58f5fe1d in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00007f9c58f5fecf in sigtimedwait () from /lib/x86_64-linux-gnu/libc.so.6
#2  0x0000000000408c05 in sighandler (x=<optimized out>) at main.c:98
#3  0x000000000040b500 in thread_detach_func (d=0x19a1480) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 5 (Thread 0x7f9c2bfff700 (LWP 14662)):
#0  0x00007f9c59012c03 in epoll_wait () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x000000000040af6b in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:302
#2  0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
#3  0x000000000040b500 in thread_detach_func (d=0x19a13a0) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 4 (Thread 0x7f9c2b7fe700 (LWP 14663)):
#0  0x00007f9c59012c03 in epoll_wait () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x000000000040af6b in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:302
#2  0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
#3  0x000000000040b500 in thread_detach_func (d=0x19a1390) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 3 (Thread 0x7f9c50924700 (LWP 14661)):
#0  0x00007f9c59012c03 in epoll_wait () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x000000000040af6b in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:302
#2  0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
#3  0x000000000040b500 in thread_detach_func (d=0x19a13b0) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 2 (Thread 0x7f9c51125700 (LWP 14660)):
#0  0x00007f9c59012c03 in epoll_wait () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x000000000040af6b in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:302
#2  0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
#3  0x000000000040b500 in thread_detach_func (d=0x19a13c0) at aux.c:143
#4  0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 1 (Thread 0x7f9c52928700 (LWP 14657)):
#0  0x00007f9c5ad43aa5 in ?? () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
#1  0x00007f9c5adb4b25 in ?? () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
#2  0x00007f9c5adaf775 in EVP_EncryptUpdate () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
#3  0x0000000000423ebd in aes_ctr_128 (out=<optimized out>, in=<optimized out>, ecc=0x7f9c2401a800, iv=iv@entry=0x7f9c52925210 "\336\323\332\r=R\233\305(Q\271\256\305\311") at crypto.c:232
#4  0x0000000000424606 in aes_cm_encrypt (c=<optimized out>, ssrc=<optimized out>, s=<optimized out>, idx=<optimized out>) at crypto.c:369
#5  0x0000000000424e73 in crypto_encrypt_rtp (index=18232, payload=0x7f9c52925250, rtp=<optimized out>, c=0x7f9c340356f0) at crypto.h:119
#6  rtp_avp2savp (s=0x7f9c52925320, c=0x7f9c340356f0) at rtp.c:141
#7  0x000000000043827d in stream_packet (tv=0x7f9c52925330, fsin=0x7f9c52925340, s=0x7f9c52925320, sfd=<optimized out>) at media_socket.c:1247
#8  stream_fd_readable (fd=<optimized out>, p=<optimized out>, u=<optimized out>) at media_socket.c:1461
#9  0x000000000040b0ee in poller_poll (p=p@entry=0x19be970, timeout=timeout@entry=100) at poller.c:348
#10 0x000000000040b48d in poller_loop (d=0x19be970) at poller.c:502
#11 0x000000000040b500 in thread_detach_func (d=0x19a1420) at aux.c:143
#12 0x00007f9c592dd064 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#13 0x00007f9c5901262d in clone () from /lib/x86_64-linux-gnu/libc.so.6
```

The rtpengine version is 5.1.0.0+0~mr5 built January 26 this year. We are running Debian Jessie, so the OpenSSL version is 1.0.1t.

The log contains nothing special, although running on log level 7. Those are the last/first lines around the last crash:

```
ov 11 10:56:35 glenne rtpengine[14650]: DEBUG: [51f5087f129148391b300b0c03b18f2c@sipgate.de port 15336]: DTLS handshake successful
Nov 11 10:56:35 glenne rtpengine[14650]: INFO: [51f5087f129148391b300b0c03b18f2c@sipgate.de port 15336]: DTLS-SRTP successfully negotiated
Nov 11 10:56:35 glenne rtpengine[14650]: DEBUG: [51f5087f129148391b300b0c03b18f2c@sipgate.de port 15336]: SRTP keys, incoming:
Nov 11 10:56:35 glenne rtpengine[14650]: DEBUG: [51f5087f129148391b300b0c03b18f2c@sipgate.de port 15336]: --- AES_CM_128_HMAC_SHA1_80 key 3Z1JbkZLJtqua9qTOFLWow== salt yDAifXinBgjCHFsrouo=
Nov 11 10:56:35 glenne rtpengine[14650]: DEBUG: [51f5087f129148391b300b0c03b18f2c@sipgate.de port 15336]: SRTP keys, outgoing:
Nov 11 10:56:35 glenne rtpengine[14650]: DEBUG: [51f5087f129148391b300b0c03b18f2c@sipgate.de port 15336]: --- AES_CM_128_HMAC_SHA1_80 key fFAsh6Mu4RFi5K4NEe5A/Q== salt xODArq92U73dQgNzGPY=
Nov 11 10:56:35 glenne rtpengine[14650]: INFO: [51f5087f129148391b300b0c03b18f2c@sipgate.de port 15336]: DTLS-SRTP successfully negotiated
Nov 11 10:56:35 glenne rtpengine[14650]: DEBUG: [51f5087f129148391b300b0c03b18f2c@sipgate.de port 15336]: SRTP keys, incoming:
Nov 11 10:56:35 glenne rtpengine[14650]: DEBUG: [51f5087f129148391b300b0c03b18f2c@sipgate.de port 15336]: --- AES_CM_128_HMAC_SHA1_80 key 3Z1JbkZLJtqua9qTOFLWow== salt yDAifXinBgjCHFsrouo=
Nov 11 10:56:35 glenne rtpengine[14650]: DEBUG: [51f5087f129148391b300b0c03b18f2c@sipgate.de port 15336]: SRTP keys, outgoing:
Nov 11 10:56:35 glenne rtpengine[14650]: DEBUG: [51f5087f129148391b300b0c03b18f2c@sipgate.de port 15336]: --- AES_CM_128_HMAC_SHA1_80 key fFAsh6Mu4RFi5K4NEe5A/Q== salt xODArq92U73dQgNzGPY=
Nov 11 10:56:35 glenne rtpengine[14650]: DEBUG: [51f5087f129148391b300b0c03b18f2c@sipgate.de port 15336]: Sending DTLS packet
Nov 11 11:01:13 glenne ngcp-rtpengine-daemon[22823]: Stopping RTP/media proxy: ngcp-rtpengine-daemon:.
Nov 11 11:01:13 glenne rtpengine[22868]: INFO: Generating new DTLS certificate
Nov 11 11:01:13 glenne rtpengine[22868]: DEBUG: Dump of DTLS certificate:
```

Is there anything you can see about this crash?

**Comment by mtryfoss** [comment]
Same as https://github.com/sipwise/rtpengine/issues/370 ?

**Comment by SipSeb** [comment]
Is there any way to verify whether it is the same problem?

**Comment by mtryfoss** [comment]
Don't really know. But we also had random crashes related to SRTP-RTP bridging for WebRTC.

It started happening after we upgraded rtpengine to a lot newer version, but I don't know which version the bug was introduced.

**Comment by SipSeb** [comment]
And did the fix mentioned in #370 help?

**Comment by mtryfoss** [comment]
Yes. No crash since.

**Comment by SipSeb** [comment]
Okay, I'll try with version 6.0.0 and see whether it helps.

**Comment by rfuchs** [comment]
I don't think this is the same crash as #370 

But looking at the top commit in branch mr5.1, I see:

commit 099ff63afe9a9fef285c93508cac9d38e4261b54 (HEAD -> mr5.1, origin/mr5.1, github/mr5.1, gerrit/mr5.1)
Author: Richard Fuchs <rfuchs@sipwise.com>
Date:   Mon Apr 10 11:08:08 2017 -0400

    TT#14008 add missing DTLS locks
    
    fixes possible segfaults due to race conditions
    
    fixes #283

So I would suggest updating to at least that one.
